libethereum/BlockChainSync.cpp

```diff
@@ -177,6 +177,12 @@ BlockChainSync::~BlockChainSync()
 	abortSync();
 }
 
+void BlockChainSync::onBlockImported(BlockHeader const& _info)
+{
+	m_lastImportedBlock = static_cast<unsigned>(_info.number());
```

Comment:
These values are not atomic and used from network-message-handling thread simultaneously with importing, so I think it should be under `RecursiveGuard l(x_sync);`